internal/unionstore/union_store.go

```diff
@@ -270,6 +250,14 @@ type MemBuffer interface {
 	FlushWait() error
 	// GetMetrics returns the metrics related to flushing
 	GetMetrics() Metrics
+
+	// GetSnapshot returns a snapshot of the MemBuffer.
+	// The snapshot returned by this function is protected by an RWLock to ensure thread safety.
+	// And this snapshot can fully replace SnapshotGetter, SnapshotIter, and SnapshotIterReverse.
+	// Additionally, it provides two iteration methods: ForEachInSnapshotRange and BatchedSnapshotIter,
+	// which tolerate interleaving reads and writes for using them simply.
+	// The snapshot also verifies the snapshot sequence number to prevent reading from an invalid snapshot.
+	GetSnapshot() MemBufferSnapshot
```

Comment:
I suggest we define the `snapshot` here.

Reply:
What do you mean by defining `snapshot` here?
The `MemBufferSnapshot` is defined in line 281 just after the `MemBuffer` interface.

Reply:
I mean define the concept in comments to explain what snapshot is

Reply:
I added some explanations about snapshot, PTAL again.